### ISSUE TITLE
som code in @elastic/apm-rum-core disappear when build

### ISSUE THREAD

**Issue by Dedicatus546** [rollup/rollup#4468]
### Rollup Version

2.70.2

### Operating System (or Browser)

windows 11

### Node Version (if applicable)

16.13.1

### Link To Reproduction

https://github.com/Dedicatus546/rollup-build-demo

### Expected Behaviour

dist/bundle.js

```javascript
function patchAll() {
  if (!alreadyPatched) {
    alreadyPatched = true;
    patchXMLHttpRequest(function (event, task) {
      patchEventHandler.send(XMLHTTPREQUEST, [event, task]);
    });
    patchFetch(function (event, task) {
      patchEventHandler.send(FETCH, [event, task]);
    });
    patchHistory(function (event, task) {
      patchEventHandler.send(HISTORY, [event, task]);
    });
    patchEventTarget(function (event, task) {
      patchEventHandler.send(EVENT_TARGET, [event, task]);
    });
  }

  return patchEventHandler;
}
```

### Actual Behaviour

dist/bundle.js

```javascript
function patchAll() {
  if (!alreadyPatched) {
    alreadyPatched = true;
    patchXMLHttpRequest(function (event, task) {
      patchEventHandler.send(XMLHTTPREQUEST, [event, task]);
    });
    patchFetch(function (event, task) {
      patchEventHandler.send(FETCH, [event, task]);
    });
    patchHistory(function (event, task) {
      patchEventHandler.send(HISTORY, [event, task]);
    });
    // patchEventTarget disappear...
  }

  return patchEventHandler;
}
```

**Comment by dnalborczyk** [comment]
thank you for the repository. @Dedicatus546 

I was able to reproduce @lukastaegert 

it seems rollup "thinks" that the callback from `patchEventTarget` is not being used or called.

**Comment by lukastaegert** [comment]
Thanks. I will have a look once I solved some issues with the commonjs plugin.

**Comment by dnalborczyk** [comment]
I tried to reduce it as much as possible, it's likely you can reduce it even further. it's also possible that this is not a reduced repro. 

https://rollupjs.org/repl/?version=2.70.2&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QiUyMHBhdGNoRXZlbnRUYXJnZXQlMjAlN0QlMjBmcm9tJTIwJy4lMkZwYXRjaEV2ZW50VGFyZ2V0LmpzJyU1Q24lNUNuZnVuY3Rpb24lMjBwYXRjaEFsbCgpJTIwJTdCJTVDbiUyMCUyMHBhdGNoRXZlbnRUYXJnZXQoZnVuY3Rpb24lMjAoZXZlbnQpJTIwJTdCJTVDbiUyMCUyMCUyMCUyMHBhdGNoRXZlbnRIYW5kbGVyLnNlbmQoJ2ZvbycpJTNCJTVDbiUyMCUyMCU3RCklM0IlNUNuJTIwJTIwcmV0dXJuJTIwcGF0Y2hFdmVudEhhbmRsZXIlM0IlNUNuJTdEJTVDbiU1Q25leHBvcnQlMjAlN0IlMjBwYXRjaEFsbCUyMCU3RCUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMnBhdGNoRXZlbnRUYXJnZXQuanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyZXhwb3J0JTIwZnVuY3Rpb24lMjBwYXRjaEV2ZW50VGFyZ2V0KGNhbGxiYWNrKSUyMCU3QiU1Q24lMjAlMjB2YXIlMjBwcm90byUyMCUzRCUyMHdpbmRvdy5FdmVudFRhcmdldC5wcm90b3R5cGUlM0IlNUNuJTIwJTIwdmFyJTIwbmF0aXZlQWRkRXZlbnRMaXN0ZW5lciUyMCUzRCUyMHByb3RvLmFkZEV2ZW50TGlzdGVuZXIlM0IlNUNuJTVDbiUyMCUyMGZ1bmN0aW9uJTIwY3JlYXRlTGlzdGVuZXJXcmFwcGVyKCklMjAlN0IlNUNuJTIwJTIwJTIwJTIwZnVuY3Rpb24lMjB3cmFwcGluZ0ZuKCklMjAlN0IlNUNuJTIwJTIwJTIwJTIwJTIwJTIwY2FsbGJhY2soKSUzQiU1Q24lMjAlMjAlMjAlMjAlN0QlNUNuJTVDbiUyMCUyMCUyMCUyMHJldHVybiUyMHdyYXBwaW5nRm4lM0IlNUNuJTIwJTIwJTdEJTVDbiU1Q24lMjAlMjBwcm90by5hZGRFdmVudExpc3RlbmVyJTIwJTNEJTIwZnVuY3Rpb24lMjAoKSUyMCU3QiU1Q24lMjAlMjAlMjAlMjB2YXIlMjB3cmFwcGluZ0xpc3RlbmVyRm4lMjAlM0QlMjBjcmVhdGVMaXN0ZW5lcldyYXBwZXIoKSUzQiU1Q24lMjAlMjAlMjAlMjByZXR1cm4lMjBuYXRpdmVBZGRFdmVudExpc3RlbmVyLmFwcGx5KHRhcmdldCUyQyUyMHdyYXBwaW5nTGlzdGVuZXJGbiklM0IlNUNuJTIwJTIwJTdEJTNCJTVDbiU1Q24lMjAlMjByZXR1cm4lMjBwcm90byU1Q24lN0QlMjIlMkMlMjJpc0VudHJ5JTIyJTNBZmFsc2UlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXMlMjIlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlN0QlMkMlMjJleGFtcGxlJTIyJTNBbnVsbCU3RA==

notice , when you remove/replace the `nativeAddEventListener` variable, it appears to work:

https://rollupjs.org/repl/?version=2.70.2&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QiUyMHBhdGNoRXZlbnRUYXJnZXQlMjAlN0QlMjBmcm9tJTIwJy4lMkZwYXRjaEV2ZW50VGFyZ2V0LmpzJyU1Q24lNUNuZnVuY3Rpb24lMjBwYXRjaEFsbCgpJTIwJTdCJTVDbiUyMCUyMHBhdGNoRXZlbnRUYXJnZXQoZnVuY3Rpb24lMjAoZXZlbnQpJTIwJTdCJTVDbiUyMCUyMCUyMCUyMHBhdGNoRXZlbnRIYW5kbGVyLnNlbmQoJ2ZvbycpJTNCJTVDbiUyMCUyMCU3RCklM0IlNUNuJTIwJTIwcmV0dXJuJTIwcGF0Y2hFdmVudEhhbmRsZXIlM0IlNUNuJTdEJTVDbiU1Q25leHBvcnQlMjAlN0IlMjBwYXRjaEFsbCUyMCU3RCUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMnBhdGNoRXZlbnRUYXJnZXQuanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyZXhwb3J0JTIwZnVuY3Rpb24lMjBwYXRjaEV2ZW50VGFyZ2V0KGNhbGxiYWNrKSUyMCU3QiU1Q24lMjAlMjB2YXIlMjBwcm90byUyMCUzRCUyMHdpbmRvdy5FdmVudFRhcmdldC5wcm90b3R5cGUlM0IlNUNuJTVDbiUyMCUyMGZ1bmN0aW9uJTIwY3JlYXRlTGlzdGVuZXJXcmFwcGVyKCklMjAlN0IlNUNuJTIwJTIwJTIwJTIwZnVuY3Rpb24lMjB3cmFwcGluZ0ZuKCklMjAlN0IlNUNuJTIwJTIwJTIwJTIwJTIwJTIwY2FsbGJhY2soKSUzQiU1Q24lMjAlMjAlMjAlMjAlN0QlNUNuJTVDbiUyMCUyMCUyMCUyMHJldHVybiUyMHdyYXBwaW5nRm4lM0IlNUNuJTIwJTIwJTdEJTVDbiU1Q24lMjAlMjBwcm90by5hZGRFdmVudExpc3RlbmVyJTIwJTNEJTIwZnVuY3Rpb24lMjAoKSUyMCU3QiU1Q24lMjAlMjAlMjAlMjB2YXIlMjB3cmFwcGluZ0xpc3RlbmVyRm4lMjAlM0QlMjBjcmVhdGVMaXN0ZW5lcldyYXBwZXIoKSUzQiU1Q24lMjAlMjAlMjAlMjByZXR1cm4lMjBwcm90by5hZGRFdmVudExpc3RlbmVyLmFwcGx5KHRhcmdldCUyQyUyMHdyYXBwaW5nTGlzdGVuZXJGbiklM0IlNUNuJTIwJTIwJTdEJTNCJTVDbiU1Q24lMjAlMjByZXR1cm4lMjBwcm90byU1Q24lN0QlMjIlMkMlMjJpc0VudHJ5JTIyJTNBZmFsc2UlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXMlMjIlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlN0QlMkMlMjJleGFtcGxlJTIyJTNBbnVsbCU3RA==

**Comment by lukastaegert** [comment]
Thanks so much for the reproduction, this saved a lot of time! Turns out a typo in Rollup's code base was the culprit (and not a fundamental flaw, phew), fix at #4486 .

**Comment by dnalborczyk** [comment]
awesome, glad it helped!
